### PR TITLE
fix: availability creation and expand/collapse settings

### PR DESCRIPTION
**Pull request by mihaic195**
- Fixes: On event-type edit, If I add an availability, each time when I expand the advanced settings and click Update, the availabilities get duplicated.
- Fixes: If I expand and then collapse the advanced settings and click Update, I get a client error due to the advanced settings visibility flag not being updated accordingly.
- Fixes: Accessing the event-type edit page throws an error if you are not the owner of a team event-type.

### CODE REVIEW

File: pages/api/availability/eventtype.ts [hunk 1]
```diff
@@ -144,6 +144,12 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
         const openingHours = req.body.availability.openingHours || [];
         // const overrides = req.body.availability.dateOverrides || [];
 
+        await prisma.availability.deleteMany({
+          where: {
+            eventTypeId: +req.body.id,
```
Comment (KATT) [inline pages/api/availability/eventtype.ts:149]: You need to make sure you are the logged in user too

```suggestion
            eventTypeId: +req.body.id,
            userId: ....
```

**Review by KATT — CHANGES_REQUESTED** [review]
this allows anyone to delete anyone's avail

File: pages/api/availability/eventtype.ts (hunk 1 above)
Comment (mihaic195) [reply pages/api/availability/eventtype.ts:149]: I don't think you will ever reach this part unless you are logged in... or am I wrong?

Comment (mihaic195) [reply pages/api/availability/eventtype.ts:149]: Indeed! Good catch!

**Review by zomars — APPROVED** [review]

**Review by emrysal — APPROVED** [review]
LGTM

File: pages/api/availability/eventtype.ts [hunk 2]
```diff
@@ -144,11 +144,13 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
...
-            eventTypeId: +req.body.id,
-          },
-        });
+        if (+req.body.id) {
+          await prisma.availability.deleteMany({
+            where: {
+              eventTypeId: +req.body.id,
+            },
+          });
+        }
```
Comment (KATT) [inline pages/api/availability/eventtype.ts:153]: declare variable

```suggestion
        const eventTypeId = +req.body.id;
        if (eventTypeId) {
          await prisma.availability.deleteMany({
            where: {
              eventTypeId,
            },
          });
        }
```

**Review by PeerRich — APPROVED** [review]

File: prisma/migrations/20211011084924_recreate_period_type_on_event_type/migration.sql [hunk 3]
```diff
@@ -0,0 +1,20 @@
+/*
+  Warnings:
+
+  - The `periodType` column on the `EventType` table would be dropped and recreated. This will lead to data loss if there is data in the column.
```
Comment (KATT) [inline prisma/migrations/20211011084924_recreate_period_type_on_event_type/migration.sql:4]: ⚠️

**Review by KATT — DISMISSED** [review]
Do not merge this with the migration file

**Review by pumfleet — APPROVED** [review]

**Pull request merged**